File: gr/demanding-studies.hs
-- demanding studies (jmcc)

import Sound.SC3.ID {- hsc3 -}

demanding_studies :: UGen
demanding_studies =
  let s1 = drand 'a' dinf (mce [72, 75, 79, 82])
      s2 = drand 'b' 1 (mce [82, 84, 86])
      s3 = dseq 'c' dinf (mce [72, 75, 79, s2])
      x = mouseX KR 5 13 Linear 0.2
      tr = impulse KR x 0
      f = demand tr 0 (mce [midiCPS (s1 - 12), midiCPS s3])
      o1 = sinOsc AR (f + mce2 0 0.7) 0
      o2 = saw AR (f + mce2 0 0.7) * 0.3
      o3 = cubed (distort (log (distort (o1 + o2))))
  in o3 * 0.1

main :: IO ()
main = audition (out 0 demanding_studies)

File: gr/demanding-studies.fs
\ demanding studies (jmcc)

5 13 1 0.2 MouseX.kr ( x )
0 Impulse.kr ( tr )
0 dinf 72 75 79 82 4 mce α Drand.dr ( s1 )
12 swap - MIDICPS dinf 72 75 79 1 82 84 86 3 mce β Drand.dr ( s2 )
4 mce γ Dseq.dr ( s3 )
MIDICPS mce2 Demand.kr ( f )
dup 0 0.7 mce2 + 0 SinOsc.ar ( o1 )
swap 0 0.7 mce2 + Saw.ar 0.3 * ( o2 )
+ Distort Log Distort Cubed 0.1 *
